src/instance.rs

```diff
@@ -39,6 +39,16 @@ pub unsafe trait PyNativeType: Sized {
     }
 }
 
+/// Trait to denote types which have a GIL-bound reference, e.g. PyAny has &'a PyAny.
+///
+/// # Safety
+/// `Py<T>` will hand out references to `Self::AsRefTarget`. This _must_ have the
+/// same layout as `*mut ffi::PyObject`.
```

Comment:
Isn't the layout part guaranteed by `PyNativeType`? Unless "this" refers to `Self` rather than the target.

Reply:
Yes, I think so. Probably still worth repeating this constraint here?

Reply:
Sure 👍